### PR TITLE
[1-23] Create Main page

### PR DESCRIPTION
**Pull request by KuznetsovR**
main page + setup for routing added

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Default ignored files
```
Comment (YanaBr) [inline .idea/.gitignore]: Please, before opening PR !always! pull changes from dev branch to resolve merge conflicts and have last changes.
Currently in dev branch .gitigrone has .idea/ and like this your .idea would not appear in the PR.
So, please, pull changes from dev, push them and remove .idea files from PR

File: exalearn-ui/.prettierrc [hunk 2]
```diff
@@ -4,12 +4,12 @@
     "endOfLine": "auto",
     "trailingComma": "none",
     "arrowParens": "always",
-    "jsxBracketSameLine": true, 
-    "useTabs": true, 
-    "tabWidth": 4, 
+    "jsxBracketSameLine": true,
+    "useTabs": true,
+    "tabWidth": 2,
```
Comment (YanaBr) [inline exalearn-ui/.prettierrc]: Why did you change it on 2?

File: exalearn-ui/package-lock.json [hunk 3]
```diff
@@ -145,65 +145,6 @@
         "rxjs": "6.6.7"
       }
     },
-    "@angular-eslint/builder": {
```
Comment (YanaBr) [inline exalearn-ui/package-lock.json]: Are you sure you want to remove all these dependencies? Did you discuss it with @HrebenYury?

File: exalearn-ui/src/app/modules/app-routing/app-routing.module.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from "@angular/router";
+import { MainPageComponent } from "../../pages/main-page/main-page.component";
+
+const routes: Routes = [
+  { path: 'main', component: MainPageComponent },
+  { path: '', redirectTo: '/main', pathMatch: 'full' },
+  // { path: 'history', component: MainPageComponent },
```
Comment (YanaBr) [inline exalearn-ui/src/app/modules/app-routing/app-routing.module.ts]: Please, remove your technical comments from code, try to store it in notepad, for instance. We leave only important for development comments, such as a description of the function and its parameters, an explanation of some of the actions, etc.

File: exalearn-ui/src/app/pages/main-page/main-page.component.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<div class="main-page">
+  <div class="btns-container">
+    <button mat-raised-button color="primary" class="btn" *ngFor="let btn of btns">{{btn.name}}</button>     <!--[routerLink]="btn.url"-->
+  </div>
+</div>
+<!--<button mat-button>Basic</button>-->
```
Comment (YanaBr) [inline exalearn-ui/src/app/pages/main-page/main-page.component.html]: Same here (look last comment about comments)

File: exalearn-ui/src/app/services/roles.service.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  private adminBtns: readonly RedirectBtn[] = [
+    {name: "statistics gathering", url: "/statistics"},
+    {name: "reassign test", url: "/reassign"},
+    {name: "reported questions", url: "/reported"}
+  ]
+
+  constructor() {
+  }
+
+  getBtns(role: "user" | "HR" | "coach" | "admin"): readonly RedirectBtn[] {
```
Comment (YanaBr) [inline exalearn-ui/src/app/services/roles.service.ts]: Please, use same styles for HR role: 'HR' -> 'hr'

File: exalearn-ui/src/app/services/roles.service.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import {Injectable} from '@angular/core';
+import {RedirectBtn} from "../interfaces/interfaces";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RolesService {
+  private userBtns: readonly RedirectBtn[] = [
+    {name: "my tests history", url: "/history"},
```
Comment (YanaBr) [inline exalearn-ui/src/app/services/roles.service.ts]: We use single quotes everywhere, please, try to refactore code with prettier.
@HrebenYury, did you configure git pre-commit hook? Or we didn't have single-quote rule?

File: exalearn-ui/src/app/services/roles.service.ts [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    {name: "pass test", url: "/test"},
```
Comment (YanaBr) [inline exalearn-ui/src/app/services/roles.service.ts]: We need to add pipe inside *ngFor to make buttons names uppercase, or at least rewrite names  with uppercase first letter: 'pass test' -> 'Pass test'

File: package-lock.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (YanaBr) [inline package-lock.json]: Do we really need this package-lock.json? I think you should delete it from PR

File: exalearn-ui/.prettierrc (hunk 2 above)
Comment (KuznetsovR) [reply exalearn-ui/.prettierrc:9]: We've discussed that we have 2 tabWidth, don't we?

Comment (KuznetsovR) [reply exalearn-ui/.prettierrc]: We have discussed that we have 2 tabwidth, don't we?

File: exalearn-ui/src/app/services/roles.service.ts (hunk 7 above)
Comment (KuznetsovR) [reply exalearn-ui/src/app/services/roles.service.ts]: changed

File: exalearn-ui/src/app/services/roles.service.ts (hunk 8 above)
Comment (KuznetsovR) [reply exalearn-ui/src/app/services/roles.service.ts]: added pipe

File: package-lock.json (hunk 9 above)
Comment (KuznetsovR) [reply package-lock.json]: deleted

File: exalearn-ui/src/app/services/roles.service.ts (hunk 6 above)
Comment (KuznetsovR) [reply exalearn-ui/src/app/services/roles.service.ts]: renamed

File: exalearn-ui/src/app/pages/main-page/main-page.component.html (hunk 5 above)
Comment (KuznetsovR) [reply exalearn-ui/src/app/pages/main-page/main-page.component.html]: also deleted

File: exalearn-ui/src/app/modules/app-routing/app-routing.module.ts (hunk 4 above)
Comment (KuznetsovR) [reply exalearn-ui/src/app/modules/app-routing/app-routing.module.ts]: comments deleted

File: exalearn-ui/package-lock.json (hunk 3 above)
Comment (KuznetsovR) [reply exalearn-ui/package-lock.json]: For some reason it was made automatically, now returned

File: exalearn-ui/src/app/services/roles.service.ts (hunk 7 above)
Comment (HrebenYury) [reply exalearn-ui/src/app/services/roles.service.ts]: > @HrebenYury, did you configure git pre-commit hook? Or we didn't have single-quote rule?

No, I didn't configure pre-commit hook. I will solve this problem as soon as possible. And we have single-quote rule into .prettierrc file.

File: exalearn-ui/.prettierrc (hunk 2 above)
Comment (YanaBr) [reply exalearn-ui/.prettierrc]: We made a call with @HrebenYury and decided that 4 is okay, it was in his PR with linter and prittier

**Review by YanaBr — APPROVED** [review]

**Review by KotdenGit — APPROVED** [review]

**Review by GeranSkot — APPROVED** [review]
